include/simdjson/implementation.h

```diff
@@ -10,6 +10,24 @@
 
 namespace simdjson {
 
+/**
+ * Validate the UTF-8 string.
+ *
+ * @param buf the string to validate.
+ * @param len the length of the string in bytes.
+ * @return true if the string is valid UTF-8.
+ */
+WARN_UNUSED bool validate_utf8(const char * buf, size_t length) noexcept;
```

Comment:
Thinking this through more: do we want to return error_code even though we only have one such error code right now? I can imagine different errors for different problems, and that would future proof it a bit. Just a question to provoke thought, I like bool a lot for simplicity and can't really decide offhand what I like better.

Also, and this is incredibly nitpicky, "length -> len" for the parameter name.

Reply:
> Also, and this is incredibly nitpicky, "length -> len" for the parameter name.

Yes! I am fixing it now.

Reply:
Fixed.

Reply:
> Thinking this through more: do we want to return error_code even though we only have one such error code right now? 

Dealing with an error_code is more mental effort than dealing with a bool, obviously.

Your proposal is of course sound, but I am sure that you are agree that adding more complexity today so that you can future proof for things that you have not even thought about... is often a bad deal.

In this instance, my mental model of the future is this...

```
if(!validate(string)) {
   tellmewhythisisbad(string);
}
```

That is, I imagine that we would use a different (slower and more complicated) function to identify the nature of the validation error.